### ISSUE TITLE
Poetry installing beta versions when it should resolver error

### ISSUE THREAD

**Issue by bertjwregeer** [sdispater/poetry#510]
- [x] I am on the [latest](https://github.com/sdispater/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.
- [ ] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: Mac OS 10.14
- **Poetry version**: 0.12.2
- `pyproject.toml`

```
[tool.poetry.dependencies]
python = "3.7"
pyramid = "~1.8.0"
pyramid_retry = "^0.5.0"
```

## Issue

I would expect this to fail resolving because `pyramid_retry` depends on `pyramid>=1.9b1` and I have pinned `pyramid` to `~1.8.0`, I would not expect it to pull in a beta version since I am not using `--allow-prerelease`.

**Comment by bertjwregeer** [comment]
Yes, we are fixing `pyramid_retry` to not depend on pre-release software, but this was unexpected.

**Comment by sdispater** [comment]
This is fixed in commit 670d31c

**Comment by sdispater** [comment]
It's fixed in the latest `0.12.3` release.
